Adding Role Mangement, Adding Creating and Delete support of projects

ashmeet13 opened the pull request:
1. Adding three levels of roles - Admin, Annotator, Manager
2. Admin and Manager can create and delete projects
3. `projects/list` will show projects only concerning the current logged in user
4. Cleaned up code a bit.

ashmeet13 committed:
Adding Role Mangement, Adding Creating and Delete support of projects

gauthamsuresh09 commented on labelit/labelit/urls.py:
```diff
@@ -15,17 +15,24 @@
+    
+    # Project Manager URL's
+    path('projects/create', login_required(create_project), name='create_project'),
+    path('projects/list', login_required(list_projects), name='projects_list'),
```
Lets make list projects as homepage

ashmeet13 replied:
Added the url for this.

gauthamsuresh09 commented on labelit/mainapp/templates/manage_projects/list.html:
```diff
@@ -2,6 +2,7 @@
 {% load static %}
 {% block extrahead %}
 <link rel="stylesheet" href="{% static 'table.css' %}">
+<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
```
this is for delete icon? i think it will be better to use a normal button instead.

ashmeet13 replied:
Removed the icon and kept simple text button.

ashmeet13 committed:
Home page load list_projects view.
Fix - Validation errors were not showing up in create project form

ashmeet13 committed:
Remove CSS import

ashmeet13 committed:
Add annotator/Remove annotator functionality added

ashmeet13 committed:
Fix - Remove annotators

Commit:
Add django crispy forms

Commit:
Control number of connection pools and connections per pool

Commit:
Add Error page

Commit:
Update base template:
- Add bootstrap js & jquery
- Fix navigation alignment

Commit:
- Modify delete project view
- Add Class for error messages
- Use ErrorMessage objects for errors
- Remove delete project & not allowed templates

Commit:
- Modify url for delete project
- Use bootstrap table for project listing
- Use bootstrap Modal for deleting project
- Remove table CSS

Commit:
Modify project creation template, using bootstrap and django crispy forms

Commit:
Modify login template to use bootstrap card and django crispy forms. Remove old CSS

ashmeet13 committed:
Merge pull request #21 from verloop/roles-patch

UI changes, migrate completely to bootstrap,  django-crispy-forms

ashmeet13 committed:
Merge add/remove under one URL - manage-annotators

Commit:
Use single selection form for managing project annotators

Commit:
Update base template

gauthamsuresh09 commented:
LGTM
